mmdet/models/dense_heads/yolo_head.py

```diff
@@ -0,0 +1,652 @@
+                self.anchor_base_sizes[i_scale],
+                device=result_raw.device,
+                dtype=torch.float32)
+
+            anchor_w = anchors[:, 0:1].view((-1, 1, 1))
+            anchor_h = anchors[:, 1:2].view((-1, 1, 1))
+
+            w_pred = torch.exp(prediction_raw[..., 2]) * anchor_w  # Width
+            h_pred = torch.exp(prediction_raw[..., 3]) * anchor_h  # Height
+
+            # bbox_pred: convert to x1y1x2y2
+            bbox_pred = torch.stack(
+                (x_center_pred - w_pred / 2, y_center_pred - h_pred / 2,
+                 x_center_pred + w_pred / 2, y_center_pred + h_pred / 2),
+                dim=3).view((-1, 4))
+            # conf and cls
+            conf_pred = torch.sigmoid(prediction_raw[..., 4]).view(-1)
+            cls_pred = torch.sigmoid(prediction_raw[..., 5:]).view(
+                -1, self.num_classes)  # Cls pred one-hot.
+
+            # Filtering out all predictions with conf < conf_thr
+            conf_thr = cfg.get('conf_thr', -1)
+            conf_inds = conf_pred.ge(conf_thr).nonzero().flatten()
+            bbox_pred = bbox_pred[conf_inds, :]
+            cls_pred = cls_pred[conf_inds, :]
+            conf_pred = conf_pred[conf_inds]
+
+            # Get top-k prediction
+            nms_pre = cfg.get('nms_pre', -1)
+            if 0 < nms_pre < conf_pred.size(0):
+                _, topk_inds = conf_pred.topk(nms_pre)
+                bbox_pred = bbox_pred[topk_inds, :]
+                cls_pred = cls_pred[topk_inds, :]
+                conf_pred = conf_pred[topk_inds]
+
+            # Save the result of current scale
+            multi_lvl_bboxes.append(bbox_pred)
+            multi_lvl_cls_scores.append(cls_pred)
+            multi_lvl_conf_scores.append(conf_pred)
+
+        # Merge the results of different scales together
+        multi_lvl_bboxes = torch.cat(multi_lvl_bboxes)
+        multi_lvl_cls_scores = torch.cat(multi_lvl_cls_scores)
+        multi_lvl_conf_scores = torch.cat(multi_lvl_conf_scores)
+
+        if multi_lvl_conf_scores.size(0) == 0:
+            return torch.zeros((0, 5)), torch.zeros((0, ))
+
+        if rescale:
+            multi_lvl_bboxes /= multi_lvl_bboxes.new_tensor(scale_factor)
+
+        # In mmdet 2.x, the class_id for background is num_classes.
+        # i.e., the last column.
+        padding = multi_lvl_cls_scores.new_zeros(multi_lvl_cls_scores.shape[0],
+                                                 1)
+        multi_lvl_cls_scores = torch.cat([multi_lvl_cls_scores, padding],
+                                         dim=1)
+
+        det_bboxes, det_labels = multiclass_nms(
+            multi_lvl_bboxes,
+            multi_lvl_cls_scores,
+            cfg.score_thr,
+            cfg.nms,
+            cfg.max_per_img,
+            score_factors=multi_lvl_conf_scores)
+
+        return det_bboxes, det_labels
+
+    @force_fp32(apply_to=('preds_raw', ))
+    def loss(self,
+             preds_raw,
```

Comment:
Use unified variable names for the same thing.

Reply:
Hi @hellock , 

Can you explain it a bit more?